python/tests/test_deserialization.py

```diff
@@ -0,0 +1,1060 @@
+        (7, Decimal("-999999999999999999999999999999999999.9999")),
+        (8, Decimal("0.00000000000000000001")),
+        (9, Decimal("42.000000")),
+        (10, Decimal("617283694")),
+        (11, Decimal("617283694.4324")),
+    ],
+)
+async def test_decimal_deserialization(
+    session: Session,
+    table_factory: TableFactory,
+    row_id: int,
+    value: str,
+):
+    table = await table_factory(
+        "id int PRIMARY KEY, price decimal",
+        "decimal_parametrized_table",
+    )
+
+    await session.execute(f"INSERT INTO {table} (id, price) VALUES ({row_id}, {value});")
+
+    result = await session.execute(f"SELECT * FROM {table} WHERE id = {row_id}")
+    row = next(result.iter_rows())
+
+    assert isinstance(row["price"], Decimal)
+    assert row["price"] == value
+
+
+# Verifies correct deserialization of CQL varint into Python int without precision loss
+@pytest.mark.asyncio
+@pytest.mark.requires_db
+@pytest.mark.parametrize(
+    "row_id,value",
+    [
+        (1, 0),
+        (2, 1),
+        (3, -1),
+        (4, 42),
+        (5, -42),
+        (6, 2**63 - 1),
+        (7, -(2**63)),
+        (8, 12345678901234567890),
+        (9, -12345678901234567890),
+        (10, 123456789012345678901234567890),
+        (11, -99999999999999999999999999999999999999),
+        (12, 10**100),
+        (13, -(10**100)),
+    ],
+)
+async def test_varint_deserialization_parametrized(
+    session: Session,
+    table_factory: TableFactory,
+    row_id: int,
+    value: str,
+):
+    table = await table_factory(
+        "id int PRIMARY KEY, val varint",
+        "varint_parametrized_table",
+    )
+
+    await session.execute(f"INSERT INTO {table} (id, val) VALUES ({row_id}, {value});")
+
+    result = await session.execute(f"SELECT * FROM {table} WHERE id = {row_id}")
+    row = next(result.iter_rows())
+
+    # Type check
+    assert isinstance(row["val"], int)
+
+    # Exact round-trip correctness
+    assert row["val"] == value
+
+
+# Verifies correct deserialization of CQL timestamp into timezone-aware datetime
+@pytest.mark.asyncio
+@pytest.mark.requires_db
+async def test_timestamp_deserialization(session: Session, table_factory: TableFactory):
+    table = await table_factory("id int PRIMARY KEY, ts timestamp", "timestamp_table")
+
+    dt1 = datetime.datetime(2024, 5, 10, 12, 30, 45, 123000, tzinfo=datetime.timezone.utc)
+    dt2 = datetime.datetime(1999, 12, 31, 23, 59, 59, 0, tzinfo=datetime.timezone.utc)
+
+    # Insert using epoch milliseconds
+    ts1 = int(dt1.timestamp() * 1000)
+    ts2 = int(dt2.timestamp() * 1000)
+
+    await session.execute(f"INSERT INTO {table} (id, ts) VALUES (1, {ts1});")
+    await session.execute(f"INSERT INTO {table} (id, ts) VALUES (2, {ts2});")
+
+    rows = list((await session.execute(f"SELECT * FROM {table}")).iter_rows())
+
+    r1 = next(r for r in rows if r["id"] == 1)
+    r2 = next(r for r in rows if r["id"] == 2)
+
+    assert r1["ts"] == dt1
+    assert r2["ts"] == dt2
+
```

Comment:
Again, please introduce regression test for edge cases. This comment applies also to other date/time related types where you used fallible conversion in deserialization.